tests/unit_tests/test_ustar_cp/test_fcNaniqr.py

```diff
@@ -0,0 +1,153 @@
+import matlab
+import numpy as np
+import pytest
+
+# Parameterized Test Cases
+@pytest.mark.parametrize(
+    "vector, expected",
+    [
+        # Case 1: Array is 3 and has no NaNs
+        (matlab.double([[1], [2], [3]]), 1.5),  # IQR of [1, 2, 3]
+
+        # Case 2: Array >3, contains NaNs, but has >3 real numbers
+        (matlab.double([[1], [2], [3], [4], [np.nan], [6]]), np.nan),
+
+        # Case 3: Array >3, contains NaNs, but has 3 or fewer real numbers
+        # Note this result is erroneous!
+        (matlab.double([[np.nan], [2], [3] ,[np.nan]]), 1.0),
+
+        # Case 4: Array >3 and has no NaNs
+        (matlab.double([[1], [2], [3], [4], [5]]), np.nan),
+    ],
+)
+def test_fcnaniqr_vector_cases(matlab_engine, vector, expected):
+    """
+    Test MATLAB's fcNaniqr function with different 1D vectors, including edge cases.
+    """
+    # Call MATLAB function
+    result = matlab_engine.fcNaniqr(vector)
+
+    # Check for None result
+    assert result is not None, "Expected non-None result from MATLAB function"
+
+    # Check if result is correct (accounting for NaNs)
+    if np.isnan(expected):
+        assert np.isnan(result), f"Expected NaN, but got {result}"
+    else:
+        assert np.isclose(result, expected), f"Expected {expected}, got {result}"
+
+@pytest.mark.parametrize(
+    "matrix, expected",
+    [
+        # Case 1a: No NaNs, Fully Populated, column length = 3
+        (matlab.double([[1, 2, 3], [4, 5, 6], [7, 8, 9]]), [np.nan, np.nan, np.nan]),
+
+        # Case 1b: No NaNs, Fully Populated, column length = 4
+        (matlab.double([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]]), [6.0, 6.0, 6.0]),
+
+        # Case 2a: Columns with NaNs, column length = 3
+        (matlab.double([[1, np.nan, 3], [4, np.nan, 6], [7, 8, 9]]), [np.nan, np.nan, np.nan]),
+
+        # Case 2b: Columns with NaNs, column length >3 (after nans ignored)
+        (matlab.double([[1, np.nan, 3], [4, np.nan, 6], [7, 8, 9], [10,11,12]]), [6.0, np.nan, 6.0]),
+
+        # Case 3: Columns with < 4 Real Rows
+        #Note weird behaviour of IQR estimation
+        (matlab.double([[1, np.nan], [2, 3]]), [1.5, np.nan]),
+
+        # Case 4: All NaNs
+        (matlab.double([[np.nan, np.nan], [np.nan, np.nan]]), [np.nan, np.nan]),
+
+        # Case 5: Different Column Lengths
+        (matlab.double([[1, 2], [4, np.nan], [7, 8], [10, 12]]), [6, np.nan]),
```

Comment:
This looks like the same columns lengths?

Reply:
yeah I see what you mean